src/components/clusterDetail/ClusterDetail.tsx

```diff
@@ -36,6 +37,28 @@ const installationSteps = [
 ];
 
 const ClusterDetail: React.FC<ClusterDetailProps> = ({ cluster }) => {
+  const [credentials, setCredentials] = React.useState<ClusterCredentialsResp>();
+  const [credentialsError, setCredentialsError] = React.useState();
+
+  const fetchCredentials = React.useCallback(() => {
+    const fetch = async () => {
+      setCredentialsError(undefined);
+      try {
+        const response = await getClusterCredentials(cluster.id);
+        setCredentials(response.data);
+      } catch (err) {
+        setCredentialsError(err);
+      }
+    };
+    fetch();
+  }, [cluster.id]);
+
+  React.useEffect(() => {
```

Comment:
We could avoid doing this by moving it into the ClusterCredentials. Is it on purpose to make the ClusterCredentials without side effects?

Reply:
we still need console URL so fetching the data in this component makes sense. useEffect without side effects is completely fine IMO